My_work_2/Least_squares_fit.py

```diff
@@ -0,0 +1,19 @@
+
+#%%
+import numpy as np
+
+#%%
+def Dij(x, y, dig = 14):
+    return np.round(np.mean(x*y) - np.mean(x) * np.mean(y), dig)
+
+def mnk_coef(x, y, dig = 6):
+    n = np.shape(x)[0]
```

Comment:
Why not x.shape[1]? What's the difference? What if x is not numpy array but just scalar?

Reply:
Same question goes for all np routine calls here. What if user input is not numpy array? How do we treat such situations?

Reply:
Is there a standard fit routine for this tusk in scipy?

Reply:
It seems that `x.shape` and `np.shape(x)` behave pretty the same and both return tuple with array dimensions. In my case x is 1D. And calling x.shape[1] would cause an error.

I supposed that my function will only deal with numpy.arrays.

I know that there is scipy.optimize.least_squares function that can kind of solve my task, but I had little time then and already had that code so I dicided to go an easy way instead of the correct one. That's my bad.

Reply:
Yeah. You are right and this methodes call the same method. yet x.shape[axis] or x.shape[0] looks a little bit more readable to me. Due to it's 1D array it would be even easier to use x.size (yet  no shape checking there).

Reply:
Regarding leasts_quares that was exactly my presumption that you've used existing code. Yet I encourage you to look for least squares in numpy (just google for numpy least squares linear fit) there is 2-liner for this. And it works in a pretty interesting way. I'd like to discuss that example (just cause I kinda like that approach)